suite-native/module-add-accounts/src/hooks/useAddCoinAccount.ts

```diff
@@ -0,0 +1,194 @@
+import { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { useNavigation } from '@react-navigation/native';
+import { A, pipe } from '@mobily/ts-belt';
+
+import { AccountType, NetworkSymbol, Network, networks } from '@suite-common/wallet-config';
+import {
+    AccountsRootState,
+    DeviceRootState,
+    LIMIT,
+    selectDevice,
+    selectDeviceAccounts,
+} from '@suite-common/wallet-core';
+import { useAlert } from '@suite-native/alerts';
+import {
+    addNetworkAccountThunk,
+    selectAreTestnetsEnabled,
+    selectDiscoverySupportedNetworks,
+    NORMAL_ACCOUNT_TYPE,
+} from '@suite-native/discovery';
+import { useTranslate } from '@suite-native/intl';
+import { useOpenLink } from '@suite-native/link';
+import {
+    RootStackParamList,
+    RootStackRoutes,
+    StackNavigationProps,
+} from '@suite-native/navigation';
+
+export const useAddCoinAccount = () => {
+    const dispatch = useDispatch();
+    const { translate } = useTranslate();
+    const openLink = useOpenLink();
+    const areTestnetsEnabled = useSelector(selectAreTestnetsEnabled);
+    const supportedNetworks = useSelector((state: DeviceRootState) =>
+        selectDiscoverySupportedNetworks(state, areTestnetsEnabled),
+    );
+    const accounts = useSelector((state: AccountsRootState & DeviceRootState) =>
+        selectDeviceAccounts(state),
+    );
+    const device = useSelector(selectDevice);
+    const { showAlert, hideAlert } = useAlert();
+    const navigation =
+        useNavigation<StackNavigationProps<RootStackParamList, RootStackRoutes.AccountDetail>>();
+    const [isAddingCoinAccount, setIsAddingCoinAccount] = useState<boolean>(false);
+
+    const supportedNetworkSymbols = pipe(
+        supportedNetworks,
+        A.map(n => n.symbol),
+        A.uniq,
```

Comment:
why `uniq`? There shoudn't be two networks with same symbol

Reply:
There are, since there is a network for every account type eg. there are btc networks for segwit, taproot, legacy etc.